pppd/auth.c

```diff
@@ -663,6 +663,8 @@ void start_link(int unit)
 	goto disconnect;
     }
 
+    the_channel->close_discovery_sock();
```

Comment:
This will crash for any channel type other than pppoe. You need to check if the_channel->close_discovery_sock is NULL.